qiskit/primitives/containers/sampler_pub.py

```diff
@@ -0,0 +1,139 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""
+Sampler Pub class
+"""
+
+from __future__ import annotations
+
+from typing import Tuple, Union
+from numbers import Integral
+
+from qiskit import QuantumCircuit
+
+from .bindings_array import BindingsArray, BindingsArrayLike
+from .shape import ShapedMixin
+
+
+class SamplerPub(ShapedMixin):
+    """Pub (Primitive Unified Bloc) for Sampler.
+
+    Pub is composed of tuple (circuit, parameter_values, shots).
+
+    If shots are provided this number of shots will be run with the sampler,
+    if ``shots=None`` the number of run shots is determined by the sampler.
+    """
+
+    __slots__ = ("_circuit", "_parameter_values", "_shots")
+
+    def __init__(
+        self,
+        circuit: QuantumCircuit,
+        parameter_values: BindingsArray | None = None,
+        shots: int | None = None,
+        validate: bool = False,
+    ):
+        """Initialize a sampler pub.
+
+        Args:
+            circuit: a quantum circuit.
+            parameter_values: a bindings array.
+            shots: A specific number of shots to run with.
+            validate: if True, the input data is validated during initialization.
```

Comment:
Sorry for the pedantry, just trying to unify the style in our V2.
```suggestion
            circuit: A quantum circuit.
            parameter_values: A bindings array.
            shots: A specific number of shots to run with.
            validate: If True, the input data is validated during initialization.
```